pkg/virtctl/vmexport/vmexport.go

```diff
@@ -501,105 +523,101 @@ func DownloadVirtualMachineExport(client kubecli.KubevirtClient, vmeInfo *VMExpo
 	if vmeInfo.PortForward {
 		stopChan, err := setupPortForward(client, vmeInfo)
 		if err != nil {
-			return err
+			return false, err
 		}
 		defer close(stopChan)
 	}
 
 	// Wait for the vmexport object to be ready
 	if err := ExportProcessingComplete(client, vmeInfo, processingWaitInterval, processingWaitTotal); err != nil {
-		return err
+		return false, err
 	}
 
 	vmexport, err := getVirtualMachineExport(client, vmeInfo)
 	if err != nil {
-		return err
+		return false, err
 	}
 	if vmexport == nil {
-		return fmt.Errorf("unable to get '%s/%s' VirtualMachineExport", vmeInfo.Namespace, vmeInfo.Name)
+		return false, fmt.Errorf("unable to get '%s/%s' VirtualMachineExport", vmeInfo.Namespace, vmeInfo.Name)
 	}
 
+	// Grab the VM Manifest and display it.
 	if vmeInfo.ExportManifest {
-		// Grab the VM Manifest and display it.
-		if err := getVirtualMachineManifest(client, vmexport, vmeInfo); err != nil {
-			return err
-		}
-	} else {
-		// Download the exported volume
-		if err := downloadVolume(client, vmexport, vmeInfo); err != nil {
-			return err
-		}
+		return getVirtualMachineManifest(client, vmexport, vmeInfo)
 	}
-	return nil
+
+	// Download the exported volume
+	return downloadVolume(client, vmexport, vmeInfo)
 }
 
-func printRequestBody(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, vmeInfo *VMExportInfo, manifestUrl string, headers map[string]string) error {
+func printRequestBody(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, vmeInfo *VMExportInfo, manifestUrl string, headers map[string]string) (bool, error) {
 	resp, err := HandleHTTPRequest(client, vmexport, manifestUrl, vmeInfo.Insecure, vmeInfo.ServiceURL, headers)
 	if err != nil {
-		return err
+		return false, err
 	}
 	defer resp.Body.Close()
 
 	// Check server response
 	if resp.StatusCode != http.StatusOK {
-		return fmt.Errorf("bad status: %s", resp.Status)
+		printToOutput("Bad status: %s\n", resp.Status)
+		return false, nil
 	}
 	bodyAll, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return err
+		return false, err
 	}
 	fmt.Fprintf(vmeInfo.OutputWriter, "%s", bodyAll)
-	return nil
+	return true, nil
 }
 
-func getVirtualMachineManifest(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, vmeInfo *VMExportInfo) error {
+func getVirtualMachineManifest(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport, vmeInfo *VMExportInfo) (bool, error) {
```

Comment:
nit: can we avoid here the bool and only rely on the error?

Reply:
That would mean either parsing the error message or retrying on every error kind (not only bad status). Having the boolean might be a bit overkill but I don't love any of the three options.